extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisReceiver.scala

```diff
@@ -216,6 +226,25 @@ private[kinesis] class KinesisReceiver[T](
     shardIdToLatestStoredSeqNum.get(shardId)
   }
 
+  /** Pass the setCheckpointer request to the checkpointer. */
+  private[kinesis] def setCheckpointerForShardId(
```

Comment:
setCheckpointer is sufficient. Scala docs should just say "set the checkpoint that will be used to checkpoint sequence numbers to DynamoDB". Whether there is a "setCheckpointer method or not on some other checkpointer - these are implementation details.